Adding oracle driver to the mix

ochaloup opened the pull request:
@jfdenise per your comment at https://github.com/jboss-container-images/jboss-eap-modules/pull/166 I'm opening this PR.

It's not finished and I want to use this as possibility to check with you what is expected to be part of the PR.
* the oracle driver part works
* the oracle datasource part does not work - as first I'm not sure how to define the classes for `valid-connection-checker/stale-connection-checker/exception-sorter`. Where I can find the repository of galleon datasource layers to check what are the options?
* the oracle datasource part uses the `connection-url` to be created from `host/port/dbname`. But I wonder if that scheme is expected or a env variable for url could be used? I think it's better fit to use url for Oracle than to build the url from other values.
* should be created two PR? One for `eap` branch and the other for `master`(wildfly)?

jfdenise commented on pom.xml:
```diff
@@ -42,6 +42,7 @@
         <version.org.wildfly>7.3.0.CD18-redhat-00002</version.org.wildfly>
         <version.org.wildfly.galleon-plugins>4.0.3.Final</version.org.wildfly.galleon-plugins>
         <version.org.postgresql.postgresql>42.2.8</version.org.postgresql.postgresql>
+        <version.ojdbc>19.3.0.0</version.ojdbc>
```
Perhaps name the variable version.ojdbc10.

jfdenise commented on src/main/resources/layers/standalone/oracle-driver/layer-spec.xml:
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" ?>
+<layer-spec xmlns="urn:jboss:galleon:layer-spec:1.0" name="oracle-driver">
```
perhaps oracle-10-driver?

jfdenise commented on src/main/resources/layers/standalone/oracle-driver/layer-spec.xml:
```diff
@@ -0,0 +1,15 @@
+<layer-spec xmlns="urn:jboss:galleon:layer-spec:1.0" name="oracle-driver">
+    <feature spec="subsystem.datasources">
+        <feature spec="subsystem.datasources.jdbc-driver">
+            <param name="driver-name" value="oracle"/>
```
perhaps oracle-10?

jfdenise commented on src/main/resources/modules/com/oracle/ojdbc/main/module.xml:
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module name="com.oracle.ojdbc" xmlns="urn:jboss:module:1.8">
```
perhaps com.oracle.ojdbc10

jfdenise commented on src/main/resources/layers/standalone/oracle-datasource/layer-spec.xml:
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" ?>
+<layer-spec xmlns="urn:jboss:galleon:layer-spec:1.0" name="oracle-datasource">
```
Should oracle10-datasource be used?

jfdenise commented:
@ochaloup , thank-you. Yes 2 PR.
I am thinking that perhaps '10' should appear in names. I am not 100% sure, I don't know how oracle handle it.
+1 on the URL env var. Perhaps this will need to be documented there: https://github.com/wildfly-extras/wildfly-datasources-galleon-pack/blob/master/README.md
To find the datasources options, do a local build of wildfly, then in galleon-pack/target/layout/org.wildfly/wildfly-galleon-pack/version/features/subsystem.datasources.datasource/spec.xml you will find all the options.

ochaloup commented:
@jfdenise ok, I hope I addressed your notes to the PR. I added the oracle datasource layer and updated the README. May you check this again, please? Thanks.
